### PR TITLE
feature(smart-contracts): deployed on base's test net

### PR DESCRIPTION
**Pull request by julien51**
# Description

Unlock has now been deployed on base's testnet.
Unfortunately contract verification is not working as expected.
I am also adding subgraphs to make sure our front-end can work.
The contract is currently owned by one of my local hot keys... we need to move to multisig shortly!

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: packages/hardhat-helpers/src/etherscan.js [hunk 1]
```diff
@@ -14,6 +14,7 @@ const etherscan = {
     polygonMumbai: 'W9TVEYKW2CDTQ94T3A2V93IX6U3IHQN5Y3',
     avalanche: 'N4AF8AYN8PXY2MFPUT8PAFSZNVJX5Q814X',
     celo: '6KBKUFYV3NQR4Y1BQN3Q34S2U7NTZBBPQZ',
+    baseGoerli: 'YourApiKeyToken',
```
Comment (julien51) [inline packages/hardhat-helpers/src/etherscan.js:17]: No API on basescan!

**Comment by julien51** [comment]
Summary of the task:

```
$ yarn run hardhat deploy --unlock-version 11 --public-lock-version 12 --network baseGoerli --weth-address 0x4200000000000000000000000000000000000006
Starting deployments on Base Goerli (Testnet)...
UNLOCK DEPLOYMENT > Deploying contracts on Base Goerli (Testnet) with the account: 0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266
UNLOCK DEPLOYMENT > isLocalNet : false
UNLOCK SETUP > Setting up version 11 from package

contracts/interfaces/IPublicLock.sol:3:1: Warning: Experimental features are turned on. Do not use experimental features on live deployments.
pragma experimental ABIEncoderV2;
^-------------------------------^

contracts/interfaces/IPublicLock.sol:690:43: Warning: This declaration shadows an existing declaration.
  function owner() external view returns (address owner);
                                          ^-----------^
contracts/interfaces/IPublicLock.sol:690:3: The shadowed declaration is here:
  function owner() external view returns (address owner);
  ^-----------------------------------------------------^

contracts/interfaces/IPublicLock.sol:696:28: Warning: This declaration shadows an existing declaration.
  ) external view returns (bool isOwner);
                           ^----------^
contracts/interfaces/IPublicLock.sol:694:3: The shadowed declaration is here:
  function isOwner(
  ^ (Relevant source part starts here and spans across multiple lines).

contracts/test-artifacts/TestEventHooks.sol:185:28: Warning: This declaration shadows an existing declaration.
  ) external view returns (bool isValidKey) {
                           ^-------------^
contracts/test-artifacts/TestEventHooks.sol:178:3: The shadowed declaration is here:
  function isValidKey(
  ^ (Relevant source part starts here and spans across multiple lines).

Warning: SPDX license identifier not provided in source file. Before publishing, consider adding a comment containing "SPDX-License-Identifier: <SPDX-License>" to each source file. Use "SPDX-License-Identifier: UNLICENSED" for non-open-source code. Please see https://spdx.org for more information.
--> contracts/test-artifacts/TestKeyManagerUpgraded.sol


Warning: Source file does not specify required compiler version! Consider adding "pragma solidity ^0.8.13;"
--> contracts/test-artifacts/TestKeyManagerUpgraded.sol


Warning: This declaration shadows an existing declaration.
    --> @unlock-protocol/contracts/dist/UnlockDiscountToken/UnlockDiscountTokenV2.sol:1724:25:
     |
1724 |     function initialize(string memory name, string memory symbol, uint8 decimals) public virtual initializer {
     |                         ^^^^^^^^^^^^^^^^^^
Note: The shadowed declaration is here:
    --> @unlock-protocol/contracts/dist/UnlockDiscountToken/UnlockDiscountTokenV2.sol:1730:5:
     |
1730 |     function name() public view virtual returns (string memory) {
     |     ^ (Relevant source part starts here and spans across multiple lines).


Warning: This declaration shadows an existing declaration.
    --> @unlock-protocol/contracts/dist/UnlockDiscountToken/UnlockDiscountTokenV2.sol:1724:45:
     |
1724 |     function initialize(string memory name, string memory symbol, uint8 decimals) public virtual initializer {
     |                                             ^^^^^^^^^^^^^^^^^^^^
Note: The shadowed declaration is here:
    --> @unlock-protocol/contracts/dist/UnlockDiscountToken/UnlockDiscountTokenV2.sol:1734:5:
     |
1734 |     function symbol() public view virtual returns (string memory) {
     |     ^ (Relevant source part starts here and spans across multiple lines).


Warning: This declaration shadows an existing declaration.
    --> @unlock-protocol/contracts/dist/UnlockDiscountToken/UnlockDiscountTokenV2.sol:1724:67:
     |
1724 |     function initialize(string memory name, string memory symbol, uint8 decimals) public virtual initializer {
     |                                                                   ^^^^^^^^^^^^^^
Note: The shadowed declaration is here:
    --> @unlock-protocol/contracts/dist/UnlockDiscountToken/UnlockDiscountTokenV2.sol:1738:5:
     |
1738 |     function decimals() public view virtual returns (uint8) {
     |     ^ (Relevant source part starts here and spans across multiple lines).


Warning: This declaration shadows an existing declaration.
    --> contracts/ERC20Patched.sol:2206:5:
     |
2206 |     string memory name,
     |     ^^^^^^^^^^^^^^^^^^
Note: The shadowed declaration is here:
    --> contracts/ERC20Patched.sol:2215:3:
     |
2215 |   function name()
     |   ^ (Relevant source part starts here and spans across multiple lines).


Warning: This declaration shadows an existing declaration.
    --> contracts/ERC20Patched.sol:2207:5:
     |
2207 |     string memory symbol,
     |     ^^^^^^^^^^^^^^^^^^^^
Note: The shadowed declaration is here:
    --> contracts/ERC20Patched.sol:2224:3:
     |
2224 |   function symbol()
     |   ^ (Relevant source part starts here and spans across multiple lines).


Warning: This declaration shadows an existing declaration.
    --> contracts/ERC20Patched.sol:2208:5:
     |
2208 |     uint8 decimals
     |     ^^^^^^^^^^^^^^
Note: The shadowed declaration is here:
    --> contracts/ERC20Patched.sol:2233:3:
     |
2233 |   function decimals() public view virtual returns (uint8) {
     |   ^ (Relevant source part starts here and spans across multiple lines).


Warning: Unnamed return variable can remain unassigned. Add an explicit return with value to all non-reverting code paths or name the variable.
  --> contracts/test-artifacts/TestERC20.sol:12:21:
   |
12 |   ) public returns (uint256) {
   |                     ^^^^^^^


Warning: Unused function parameter. Remove or comment out the variable name to silence this warning.
    --> @unlock-protocol/contracts/dist/UnlockDiscountToken/UnlockDiscountTokenV2.sol:1001:43:
     |
1001 |     function __ERC20Permit_init_unchained(string memory name) internal initializer {
     |                                           ^^^^^^^^^^^^^^^^^^


Warning: Unused function parameter. Remove or comment out the variable name to silence this warning.
    --> @unlock-protocol/contracts/dist/UnlockDiscountToken/UnlockDiscountTokenV2.sol:1005:40:
     |
1005 |     function __ERC20Permit_init_unsafe(string memory name) internal {
     |                                        ^^^^^^^^^^^^^^^^^^


Warning: Unused function parameter. Remove or comment out the variable name to silence this warning.
    --> contracts/ERC20Patched.sol:1200:5:
     |
1200 |     string memory name
     |     ^^^^^^^^^^^^^^^^^^


Warning: Unused function parameter. Remove or comment out the variable name to silence this warning.
    --> contracts/ERC20Patched.sol:1208:5:
     |
1208 |     string memory name
     |     ^^^^^^^^^^^^^^^^^^


Warning: Unused function parameter. Remove or comment out the variable name to silence this warning.
    --> contracts/UnlockProtocolGovernor.sol:3210:5:
     |
3210 |     string memory name
     |     ^^^^^^^^^^^^^^^^^^


Warning: Function state mutability can be restricted to pure
 --> contracts/test-artifacts/TestERC721Receiver.sol:5:3:
  |
5 |   function onERC721Received(
  |   ^ (Relevant source part starts here and spans across multiple lines).


Compiled 117 Solidity files successfully
Warning: Potentially unsafe deployment of contracts/past-versions/UnlockV11.sol:Unlock

    You are using the `unsafeAllow.delegatecall` flag.

UNLOCK SETUP > Unlock proxy deployed to: 0x51A1ceB83B83F1985a81C295d1fF28Afef186E02 (tx: 0xf38bf59b2ac92fe28fb793c22fa4cd73e067dca682538499f7a593da437273e8)  - implementation at: 0x1fA02b2d6A771842690194Cf62D91bdd92BfE28d
PUBLIC LOCK > Deploying lock template on Base Goerli (Testnet) for released version 12 with signer 0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266
PUBLIC LOCK > deployed v12 to : 0x36b58F5C1969B7b6591D752ea6F5486D069010AB (tx: 0xd83a7dcd308d7e0f4dac5b4c5550a05d5c5113069ebc9b88218da04a164d8f17)
LOCK TEMPLATE SETUP > Setting up PublicLock version 12
UNLOCK SETUP> Template set for Lock (tx: 0x1839c41c05eaa35c130ffe4992c1334cd7c96de103b3f7cc535b73bfc4d9c178)
UNLOCK CONFIG > Configuring Unlock with the following:
      - udtAddress: 0x0000000000000000000000000000000000000000
      - wethAddress: 0x4200000000000000000000000000000000000006
      - estimatedGasForPurchase: 200000
      - symbol: KEY
      - locksmithURI: https://locksmith.unlock-protocol.com/api/key/84531/
      - chainId: 84531
  
UNLOCK CONFIG > Unlock configured properly. (tx: 0x548e3410b6fdf65936df9df082fe25f63c34b5a5f8ad9579376ed5870d134448)
LOCK DEPLOY > deployed to : 0xF380bC8b4e595dECa3A55A4C98A6C4fA1c96f537 (tx: 0xfb2ec177e58832b6138f21fef1a74d64bc6e0fe399673f6506dcee92565ab126) 
 with params: 2592000,0x0000000000000000000000000000000000000000,10000000000000000,100,Test Lock
```

File: packages/hardhat-helpers/src/etherscan.js (hunk 1 above)
Comment (clemsos) [reply packages/hardhat-helpers/src/etherscan.js:17]: is that using blockscout?

**Review by clemsos — APPROVED** [review]
LG

**Pull request merged**